src/box/sql/build.c

```diff
@@ -290,39 +274,25 @@ sql_shallow_space_copy(struct Parse *parse, struct space *space)
 {
 	assert(space->def != NULL);
 	struct space *ret = sql_template_space_new(parse, space->def->name);
-	if (ret == NULL)
-		goto error;
 	ret->index_count = space->index_count;
 	ret->index_id_max = space->index_id_max;
-	size_t size = 0;
-	ret->index = region_alloc_array(&parse->region, typeof(struct index *),
-					ret->index_count, &size);
-	if (ret->index == NULL) {
-		diag_set(OutOfMemory, size, "region_alloc_array", "ret->index");
-		goto error;
-	}
+	ret->index = xregion_alloc_array(&parse->region, typeof(struct index *),
+					 space->index_count);
+	size_t size = sizeof(struct index) * space->index_count;
 	memcpy(ret->index, space->index, size);
```

Comment:
Nit: I wouldn't introduce the local variable `size` because it's used only once in `memcpy` - I'd pass the size directly to `memcpy` instead. Same comment to the `fields_size`.

Reply:
Fixed.